analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Nullable.cs

```diff
@@ -192,4 +192,70 @@ public void Nullable_Conversion_PropagateConstraints()
         validator.ValidateTag("ToNullableAs", x => x.HasConstraint(BoolConstraint.True).Should().BeTrue());
         validator.ValidateTag("ToBoolExplicit", x => x.HasConstraint(BoolConstraint.True).Should().BeTrue());
     }
+
+    [TestMethod]
+    public void Nullable_GetValueOrDefault_Int()
+    {
+        const string code = """
+            var value = arg.GetValueOrDefault();
+            Tag("UnknownArg", arg);
+            Tag("UnknownValue", value);
+
+            arg = null;
+            value = arg.GetValueOrDefault();
+            Tag("NullArg", arg);
+            Tag("NullValue", value);
+
+            arg = 42;
+            value = arg.GetValueOrDefault();
+            Tag("NotNullArg", arg);
+            Tag("NotNullValue", value);
+            """;
+        var validator = SETestContext.CreateCS(code, ", int? arg").Validator;
```

Comment:
Add a DummyContraint, so we can see when a copy of the constraints occurs.

Reply:
This is visible in `Nullable_GetValueOrDefault_Bool` with `True` instead.

Reply:
That's kind of right. It only misses the part where `arg = null` leads to the default logic and loosing info about `arg`. On the other hand: If arg is null how likely is it that there are other constraints as well? But it would still be useful to document.